csrc/ir_nodes.cpp

```diff
@@ -1329,6 +1329,253 @@ Val* GroupedWelfordOp::getInitValOfOutput(Val* output_val) const {
 
 NVFUSER_DEFINE_CLONE_AND_CREATE(GroupedWelfordOp)
 
+//==============================================================================================================================
+
+// MmaOp utils
+namespace MmaOpUtils {
+
+// The expected number of concrete domains for gemm
+constexpr size_t expected_gemm_cdomains = 2;
+
+// A helper structure used to gather all data created during analysis
+struct MmaOpDetails {
+  using AxesData = MmaOp::AxesData;
+  // Concrete axes from A that are broadcast in B and are not
+  //  reduction in output
+  AxesData m_axes;
+  // Concrete axes from B that are broadcast in A and are not
+  //  reduction in output
+  AxesData n_axes;
+  // Concrete axes from A that are concrete in B and are
+  //  reduction in output
+  AxesData k_axes;
+  // Anything that is not in m/n/k axes
+  AxesData batch_axes;
```

Comment:
There is a gray zone here: what if there is an axis that is concrete in `A`, broadcast in `B`, and reduced in output? I don't think this axis should be classified as `batch`. But on the other hand, I don't think it is `k` either. Should we raise an error if such axis exist?

Reply:
You are right, batch is currently 'anything that don't fit into defined rules' which is also not good. For batches axes we know that the are concrete in all tensors `A`,`B`,`out` (can be implemented similar way as `getMOrNaxes`. Then the check can be simply adding `M/N/K/Batches` together and check if the number of axes match the number of domains in Tensor view.

I will check this and if there are no issues, I will update accordingly.
Thanks for this pointing out!

Reply:
I think they don't have to be concrete in all tensors. For example, we can have A with shape `(B, M, K)` and B with shape `(1, K, N)`, then this is one style of "batched" matmul where only A is batched.

Reply:
I think bath axes should either concrete in all tensors, or broadcast in all tensors. For the case of A `(1, M, K)` and B `(1, K, N)`, the broadcast dimension `1` should be classified as batch.

Reply:
I updated implementation so we will mark as batch axis if:
- in all tensors (inputs/output) this axis is concrete,
- in all tensors (inputs/output) this axis is marked as broadcast,

Thanks for the suggestion!